### ISSUE TITLE
Add tabs to search results

### ISSUE THREAD

**Issue by SzymonSporysz** [marverenic/Jockey#288]

**Comment by marverenic** [comment]
I'm pretty partial to the mixed results page that's currently there. I can see where it would be convenient if you only care about a specific type of result, but I like being able to see results from all types when you've got a more specific query is more convenient. The current search page also lets you just use the keyboard to find what you're looking for, without having to switch between tabs.

Maybe a "top results" section would be more helpful? Just something that would highlight a couple of results that best match what you're searching for (regardless of type).

**Comment by SzymonSporysz** [comment]
First of all, try changing the results order to Artists, Albums, Songs. I can't really tell about the top results thing - I guess it depends on how well it'd work.

**Comment by SzymonSporysz** [comment]
Instead of doing that, maybe just display only direct results? So if a user searches for an artist, don't show all their songs and albums. If an album - don't show the songs it contains.

**Comment by marverenic** [comment]
Right now searching includes a song if either its song name, artist name, or album name contains the search query (and includes an album if either its album name or artist name contain the query). Just to confirm, you're saying to change this so that songs and albums are only included in the results if their name contains the query, right?

**Comment by SzymonSporysz** [comment]
Yes, instead of creating a seperate section with best results, I'd narrow all results shown to the best, accurate ones. This way if you only reorder elements to Artists, Albums, Songs it's much quicker for users to find what they want to.
